store/members/members-context.js

```diff
@@ -0,0 +1,27 @@
+import { createContext, useContext, useReducer } from 'react';
+import { membersReducer } from './members-reducer';
+import PropTypes from 'prop-types';
+
+const MembersContext = createContext({});
+const initialState = {
+  membersArr: [],
+  newMembersArr: [],
+  errorMsg: ''
+};
+
+export const MembersProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(membersReducer, initialState);
+  return <MembersContext.Provider value={{ state, dispatch }}>{children}</MembersContext.Provider>;
+};
+
+export const useMembers = () => {
```

Comment:
why are we naming this as `useMembers`? you know right in react if name something like `useSomeRandomTExt` it's considered as a custom hook and all the rules of hooks are need to be followed